openapps/opencoap/opencoap.c

```diff
@@ -276,48 +302,49 @@ void opencoap_receive(OpenQueueEntry_t* msg) {
       temp_desc = opencoap_vars.resources;
       
       // iterate until matching resource found, or no match
-      while (found==FALSE && securityReturnCode==0) {
-         if (
-               coap_incomingOptions[0].type==COAP_OPTION_NUM_URIPATH    &&
-               coap_incomingOptions[1].type==COAP_OPTION_NUM_URIPATH    &&
-               temp_desc->path0len>0                                   &&
-               temp_desc->path0val!=NULL                               &&
-               temp_desc->path1len>0                                   &&
-               temp_desc->path1val!=NULL
-            ) {
-            // resource has a path of form path0/path1
-               
-            if (
-                  coap_incomingOptions[0].length==temp_desc->path0len                               &&
-                  memcmp(coap_incomingOptions[0].pValue,temp_desc->path0val,temp_desc->path0len)==0 &&
-                  coap_incomingOptions[1].length==temp_desc->path1len                               &&
-                  memcmp(coap_incomingOptions[1].pValue,temp_desc->path1val,temp_desc->path1len)==0
-               ) {
-               if (temp_desc->securityContext != NULL && 
-                   blindContext != temp_desc->securityContext) {
-                   securityReturnCode = COAP_CODE_RESP_UNAUTHORIZED;
-               }
-                found = TRUE;
-            };
-         
-         } else if (
-               coap_incomingOptions[0].type==COAP_OPTION_NUM_URIPATH    &&
-               temp_desc->path0len>0                            &&
-               temp_desc->path0val!=NULL
-            ) {
-            // resource has a path of form path0
-               
-            if (
-                  coap_incomingOptions[0].length==temp_desc->path0len                               &&
-                  memcmp(coap_incomingOptions[0].pValue,temp_desc->path0val,temp_desc->path0len)==0
-               ) {
-               if (temp_desc->securityContext != NULL && 
-                   blindContext != temp_desc->securityContext) {
-                   securityReturnCode = COAP_CODE_RESP_UNAUTHORIZED;
-               }
-               found = TRUE;
-            };
-         };
+      while (found==FALSE && securityReturnCode==COAP_CODE_EMPTY) {
+        
+        option_count = opencoap_find_option(coap_incomingOptions, coap_incomingOptionsLen, COAP_OPTION_NUM_URIPATH, &option_index);
+        if (
+             option_count >= 2                                       &&
```

Comment:
by today only path0/path1 is provided. i agree checking for `option_count == 2` in others cases no matching resource is found and we return a msg with `coap_header.Code = COAP_CODE_RESP_NOTFOUND;`